dbt/adapters/spark/impl.py

```diff
@@ -364,6 +364,30 @@ def get_rows_different_sql(
 
         return sql
 
+    # This is for use in the test suite
+    # Spark doesn't have 'commit' and 'rollback', so this override
+    # doesn't include those commands.
+    def run_sql_for_tests(self, sql, fetch, conn):
+        cursor = conn.handle.cursor()
+        try:
+            cursor.execute(sql)
+            if fetch == "one":
+                if hasattr(cursor, 'fetchone'):
+                    return cursor.fetchone()
+                else:
+                    # AttributeError: 'PyhiveConnectionWrapper' object has no attribute 'fetchone'
+                    return cursor.fetchall()[0]
+            elif fetch == "all":
+                return cursor.fetchall()
+            else:
+                return
+        except BaseException as e:
+            print(sql)
```

Comment:
When should we use print and when should we use logger? Not a blocker for this issue but trying to understand our habit.

Reply:
Normally in mainline code we should use the logger. This is just for tests, and has been that way for ages, so this is just copying the way it worked already. It might be nice to turn it into regular logging at some point. We do have a separate file for test "events": core/dbt/events/test_types.py. It really is hard to page through the test logs when debugging sometimes, so it would be nice to make that easier (if we can).